zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -94,160 +127,166 @@ private DirectBuffer getValue(
-    columnFamilyContext.wrapValueView(new byte[0]);
+  public void deleteExisting(final KeyType key) {
     ensureInOpenTransaction(
-        context,
         transaction -> {
           columnFamilyContext.writeKey(key);
-          getValue(context, columnFamilyContext);
+          assertExists(transaction);
+          transaction.delete(
+              transactionDb.getDefaultNativeHandle(),
+              columnFamilyContext.getKeyBufferArray(),
+              columnFamilyContext.getKeyLength());
         });
-    return !columnFamilyContext.isValueViewEmpty();
   }
 
   @Override
-  public boolean isEmpty() {
-    final AtomicBoolean isEmpty = new AtomicBoolean(true);
-    whileEqualPrefix(
-        context,
-        keyInstance,
-        valueInstance,
-        (key, value) -> {
-          isEmpty.set(false);
-          return false;
+  public void deleteIfExists(final KeyType key) {
+    ensureInOpenTransaction(
+        transaction -> {
+          columnFamilyContext.writeKey(key);
+          transaction.delete(
+              transactionDb.getDefaultNativeHandle(),
+              columnFamilyContext.getKeyBufferArray(),
+              columnFamilyContext.getKeyLength());
         });
-    return isEmpty.get();
   }
 
-  public void forEach(final TransactionContext context, final Consumer<ValueType> consumer) {
-    whileEqualPrefix(
-        context,
-        keyInstance,
-        valueInstance,
-        (BiConsumer<KeyType, ValueType>) (ignore, value) -> consumer.accept(value));
+  @Override
+  public boolean exists(final KeyType key) {
+    ensureInOpenTransaction(
+        transaction -> {
+          columnFamilyContext.writeKey(key);
+          final byte[] value =
+              transaction.get(
+                  transactionDb.getDefaultNativeHandle(),
+                  transactionDb.getReadOptionsNativeHandle(),
+                  columnFamilyContext.getKeyBufferArray(),
+                  columnFamilyContext.getKeyLength());
+          columnFamilyContext.wrapValueView(value);
+        });
+    return !columnFamilyContext.isValueViewEmpty();
   }
 
-  public void forEach(
-      final TransactionContext context, final BiConsumer<KeyType, ValueType> consumer) {
-    whileEqualPrefix(context, keyInstance, valueInstance, consumer);
+  @Override
+  public boolean isEmpty() {
+    final AtomicBoolean isEmpty = new AtomicBoolean(true);
+    ensureInOpenTransaction(
+        transaction ->
+            forEachInPrefix(
+                new DbNullKey(),
+                (key, value) -> {
+                  isEmpty.set(false);
+                  return false;
+                }));
+
+    return isEmpty.get();
   }
 
-  public void whileTrue(
-      final TransactionContext context, final KeyValuePairVisitor<KeyType, ValueType> visitor) {
-    whileEqualPrefix(context, keyInstance, valueInstance, visitor);
+  private void assertDoesNotExist(final ZeebeTransaction transaction) throws Exception {
```

Comment:
I haven't changes that yet since I wanted to follow the newly added conventions, namely that the public methods must call `ensureInOpenTransaction` and the private ones do not. I feel like the one call to `transaction.get(...)` is an okay amount of code duplication. Let me know what you think.